src/coord/src/coord.rs

```diff
@@ -5731,12 +5867,17 @@ mod timeline {
             }
         }
 
-        /// Acquire a new timestamp for writing.
+        /// Peek the current value of the write timestamp.
         ///
-        /// This timestamp will be strictly greater than all prior values of
-        /// `self.read_ts()`, and less than or equal to all subsequent values of
-        /// `self.read_ts()`.
-        pub fn write_ts(&mut self) -> T {
+        /// No operations should be assigned to the timestamp returned by this function. The
+        /// timestamp returned should only be used to compare the progress of the `TimestampOracle`
+        /// against some external source of time.
+        ///
+        /// Subsequent values of `self.read_ts()` and `self.write_ts()` will be greater or equal to
+        /// this timestamp.
+        ///
+        /// NOTE: This can be removed once DDL is included in group commits.
+        pub fn peek_write_ts(&self) -> T {
```

Comment:
@mjibson I know we talking about not doing this for the peek, but the more I think about this, the more I think it's ok. Though it's very possible I'm just missing something. The scenarios are:
- The `next()` is ahead of the current `ts` value in which case we will not sleep and we will get the same value (or slightly higher if some time passes) from `write_ts()`.
- The `next()` is behind the current `ts` value in which case we will sleep and get a new timestamp when we wake up. 
There is some possibility that the system clock shifts in between the call to this and the call to `write_ts()`, but the only issues I can think of are unnecessarily sleeping or having to wait for the next write to sleep.

Reply:
I think it was clearer before, and we've now delved into scary changes of the oracle that are easy to get wrong. Let's analyze each sentence of the doc comments:

> Peek the current value of the write timestamp

But what this function does is return what *might* be the next value of `write_ts()`. So it's not returning the current value, and "write timestamp" is not a defined term. It doesn't even guarantee that an immediate subsequent call to write_ts will be the same, because the 1ms boundary could have been passed during those intervening nanoseconds.

> No operations should be assigned to the timestamp returned by this function.

Yet `write_ts` calls this function and assigns values that were returned.

> The timestamp returned should only be used to compare ... against some external source of time.

The oracle doesn't respect that either.

> Subsequent values of `self.read_ts()` and `self.write_ts()` will be greater or equal to this timestamp.

Calling this function and then `read_ts()` will return a value less than the timestamp returned by this function if the oracle was in the reading state, so this is now wrong too.

So I think all of the sentences are wrong now. I'm aware that all of the prose was from the previous version of this function, and so it's super understandable that it's now out of date. But figuring out exactly what we can correctly say in the doc comments here is hard. I'm not convinced this is better, just because of the slight amount of misdirection added to write_ts. Also I'm unclear of the benefit this provides us compared to the previous version.

Reply:
Yeah, the docs are completely off now. My idea with this method was "If you performed a write right now, what timestamp would it be assigned?" This gives us a slightly more accurate (by 1 ms) notion of how long we need to sleep for, if we need to sleep.

Though I buy the argument that the complexity added from this is not worth a 1 ms increase in accuracy. So I think I'll just revert it to what it was previously.

Reply:
Just pushed the simplification